.github/workflows/teamsync_check_for_changes.yml

```diff
@@ -0,0 +1,54 @@
+name: teamSync check for changes
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  check-permissions:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+          fetch-depth: 0
+      
+    - name: Set up JDK
+      uses: actions/setup-java@v4
+      with:
+        java-version: '21'
+        distribution: 'adopt'
+        
+    - name: Build project
+      run: mvn clean install
+      
+    - name: Check for yml changes
+      id: files
+      run: |
+        echo "Checking for changes in permissions/*.yml files..."
+        FILES=$(git diff --name-only ${{ github.event.before }} ${{ github.sha }} -- 'permissions/*.yml')
```

Comment:
This relies on every single execution being successful (assuming no infra trouble ever), which seems unsafe.

Reply:
There's a sweeper that runs once per week, I umm'ed and erred on this. Only operating on changes files will certainly be quicker, but why re-invent the wheel instead of using something like terraform which can maintain the state?

Reply:
its a standard infra as code tool, rather than re-implementing the wheel, it allows more people to collaborate especially from e.g. the infra team.